owncloudApp/src/main/java/com/owncloud/android/presentation/files/createfolder/CreateFolderDialogFragment.kt

```diff
@@ -82,12 +87,48 @@ class CreateFolderDialogFragment : DialogFragment() {
         alertDialog.setOnShowListener {
             val okButton = alertDialog.getButton(AlertDialog.BUTTON_POSITIVE)
             okButton.isEnabled = isButtonEnabled
+
+            okButton.setOnClickListener {
+                var fileName: String = inputText.text.toString()
+                if (fileName.isEmpty()) {
+                    error = getString(R.string.uploader_upload_text_dialog_filename_error_empty)
+                } else {
+                    null
+                }
+                if (error != null) {
+                    inputLayout.isErrorEnabled = true
+                    inputLayout.error = error
+                } else {
+                    createFolderListener.onFolderNameSet(fileName, parentFolder)
+                    dialog?.dismiss()
+                }
+            }
         }
 
         inputText.doOnTextChanged { text, _, _, _ ->
             val okButton = alertDialog.getButton(AlertDialog.BUTTON_POSITIVE)
-            isButtonEnabled = !text.isNullOrBlank()
-            okButton.isEnabled = isButtonEnabled
+
+            if (text.isNullOrBlank()) {
+                okButton.isEnabled = false
+            } else if (text.length > MAX_FILENAME_LENGTH) {
+                error = String.format(
+                    getString(R.string.uploader_upload_text_dialog_filename_error_length_max),
+                    MAX_FILENAME_LENGTH
+                )
+            } else if (forbiddenChars.any { text.contains(it) }) {
+                error = getString(R.string.filename_forbidden_characters_from_server)
+            } else {
```

Comment:
Here we should disable the OK button as well

Reply:
The behavior is ok because after the` if` we validate if the `error `is not null. When it has a value okButton is disable.